core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/query/QueryValidator.java

```diff
@@ -83,7 +84,8 @@ protected Result<Void> isValid(String path) {
 
             // cannot query on extensible (=Map) types
             if (type == Map.class) {
-                return Result.failure("Querying Map types is not yet supported");
+                return JsonPath.isPathDefinite(path) ? Result.success() :
```

Comment:
please do not add an external library without detailing the choice first in a decision record.
In my opinion we cannot accept to use this method because our query model does not follow the JsonPath specs (e.g. we're not handling square brackets for indexes`[]`).
A dead simple validation would be to avoid whitespaces in the leftOperand but in every case is better to address this in a different issue+PR

Reply:
To second what @ndr-brt said, I am also against introducing JsonPath into the runtime.

Reply:
I changed it now. Only number, alphabets, ', /, : are allowed now via regex.